x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/TransformExtension.java

```diff
@@ -20,4 +21,6 @@ public interface TransformExtension {
      * source settings.
      */
     Settings getTransformDestinationIndexSettings();
+
```

Comment:
I just implement the approach sketched in https://github.com/elastic/ml-team/issues/1061, but I'm wondering whether that's ideal.

To prevent Elasticsearch and Serverless diverging by overriding code, isn't it better to have a configuration setting (e.g. `xpack.transform.min_frequency`) and set that to a different value for Serverless? @droberts195 WDYT?

Reply:
We wanted to avoid using settings that are available in stateful for customising serverless, because then it's inevitable that people will use them in stateful as well, and the more combinations of settings that are possible the less likely it is that we'll be able to test all the possible interactions.

For each small thing, like what this PR is changing, the effect of adding one more setting doesn't seem too bad. But there are hundreds of small things like this that need changing for serverless, and if we had hundreds of extra settings then that would become hard to control.

So even though it's a bit more work to use the extension class I think it's worth it to avoid a proliferation of settings.

Reply:
TBH, I think the setting isn't necessarily specific to serverless. For stateful this setting might be useful as well (in case you have a low index refresh rate).

Reply:
after offline discussion: we'll keep it like this, because we don't want more user settings